sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -35,7 +35,7 @@ def decode_cloud_event(self, cloud_event, **kwargs): # pylint: disable=no-self-u
             cloud_event = CloudEvent._from_json(cloud_event, encode) # pylint: disable=protected-access
             deserialized_event = CloudEvent._from_generated(cloud_event) # pylint: disable=protected-access
             CloudEvent._deserialize_data(deserialized_event, deserialized_event.type) # pylint: disable=protected-access
-            return deserialized_event
+            return cast(CloudEvent, deserialized_event)
```

Comment:
What was the error that triggered the need for this cast? I would expect the return type of CloudEvent._from_generated to be CloudEvent , and then this cast unecessary

Reply:
This was unnecessary - removed it